homeassistant/components/ambient_station/__init__.py

```diff
@@ -471,42 +221,39 @@ def __init__(
         ambient: AmbientStation,
         mac_address: str,
         station_name: str,
-        sensor_type: str,
-        sensor_name: str,
-        device_class: str | None,
+        description: EntityDescription,
     ) -> None:
         """Initialize the sensor."""
         self._ambient = ambient
-        self._attr_device_class = device_class
         self._attr_device_info = {
             "identifiers": {(DOMAIN, mac_address)},
             "name": station_name,
             "manufacturer": "Ambient Weather",
         }
-        self._attr_name = f"{station_name}_{sensor_name}"
+        self._attr_name = f"{station_name}_{description.name}"
         self._attr_should_poll = False
```

Comment:
Since this is constant, it can be defined as class attribute.
```py
class AmbientWeatherEntity(Entity):
    """Define a base Ambient PWS entity."""

    _attr_should_poll = False

    def __init__(self, ...) -> None: ...
```